lib/python/ensembl/brc4/runnable/download_assembly_data.py

```diff
@@ -160,20 +169,16 @@ def download_files(accession: str, dl_dir: str) -> None:
         for (ftp_dir, entry) in f.mlsd():
             if re.match(accession, ftp_dir):
                 f.cwd(ftp_dir)
-                
+
                 # Get all the files
-                for (ftp_file, file_entry) in f.mlsd():
-                    if re.match(r'^\.', ftp_file):
-                        continue
-                    if file_entry["type"] == "dir":
-                        continue
-
-                    # Copy the file locally
-                    local_path = os.path.join(dl_dir, ftp_file)
-                    with open(local_path, 'wb') as fp:
-                        f.retrbinary("RETR " + ftp_file, fp.write)
-
-    def get_files_selection(self, dl_dir: str) -> Dict[str, str]:
+                for filename in f.nlst():
+                    # Copy selected file locally
+                    if any(filename.endswith(end) and "_from_" not in filename for end in file_ends):
+                        local_path = os.path.join(dl_dir, filename)
+                        with open(local_path, 'wb') as fp:
+                            f.retrbinary("RETR " + filename, fp.write)
+
+    def get_files_selection(self, dl_dir: str, file_ends: dict) -> Dict[str, str]:
```

Comment:
Not to be fixed in this PR, but we need to be consistent in the codebase: is `dict` enough type hint, or do we want to go to the extent of `Dict[str, str]` *everywhere*?